src/relations/natsclient_provider.py

```diff
@@ -0,0 +1,96 @@
+"""NATS Client Provider for `nats` interface."""
+import ipaddress
+import logging
+
+from ops import JujuVersion, Object, SecretNotFoundError, StoredState
+
+logger = logging.getLogger(__name__)
+
+
+class NATSClientProvider(Object):
+    """`nats` interface for the client to NATS."""
+
+    state = StoredState()
+
+    def __init__(self, charm, relation_name, listen_on_all_addresses, client_port):
+        super().__init__(charm, relation_name)
+        self._relation_name = relation_name
+        self._client_port = client_port
+        self._listen_on_all_addresses = listen_on_all_addresses
+        if listen_on_all_addresses:
+            # This will create a listening socket for all IPv4 and IPv6 addresses.
+            self._listen_address = ipaddress.ip_address("0.0.0.0")
```

Comment:
As above: shouldn't this `ipaddress.ip_address("::")` to listen for both IPv4 and IPv6?
As-is, I think this only listens on all IPv4 addresses. WDYT?